### PR TITLE
Feature/min max date

### PR DESCRIPTION
**Pull request by cs8898**
Implemented setMinDate and setMaxDate like described at #32

### CODE REVIEW

**Comment by cs8898** [comment]
The swipe code is modified to get stuck at the origins of the preferred range

if today is out of this range the init origin is forced to the corresponding min or max.

If today is after maxDate origin is set to the corresponding maxDate or the otherway round

Also added Sample wich limites the range to the current month

File: sample/src/main/res/layout/activity_main.xml [hunk 1]
```diff
@@ -34,4 +34,11 @@
       android:text="@string/title_activity_constraint"
       style="@style/Button"/>
 
+    <Button
```
Comment (thellmund) [inline sample/src/main/res/layout/activity_main.xml]: Maybe call it ”Basic Example (with limited date range)” to be consistent with the other naming and move it up a position.

File: library/src/main/java/com/alamkanak/weekview/WeekViewDrawingConfig.java [hunk 2]
```diff
@@ -191,6 +196,23 @@ void moveCurrentOriginIfFirstDraw(WeekViewConfig config) {
...
+        if (config.maxDate != null) {
+            Calendar date = (Calendar) config.maxDate.clone();
+            date.add(Calendar.DAY_OF_YEAR,1-config.numberOfVisibleDays);
+            minX = DateUtils.getXOriginForDate(date, config.getTotalDayWidth());
+        }
+        float maxX = Integer.MAX_VALUE;
+        if (config.minDate != null) {
+            maxX = DateUtils.getXOriginForDate(config.minDate, config.getTotalDayWidth());
+        }
+        if (currentOrigin.x > maxX) {
```
Comment (thellmund) [inline library/src/main/java/com/alamkanak/weekview/WeekViewDrawingConfig.java]: You can work with `Math.min()` and `Math.max()` here.

File: library/src/main/java/com/alamkanak/weekview/WeekViewDrawingConfig.java [hunk 3]
```diff
@@ -191,6 +196,23 @@ void moveCurrentOriginIfFirstDraw(WeekViewConfig config) {
 
             currentOrigin.x += (widthPerDay + config.columnGap) * difference;
         }
+
+        //HACKY Workaround for LIMIT
+        float minX = Integer.MIN_VALUE;
+        if (config.maxDate != null) {
```
Comment (thellmund) [inline library/src/main/java/com/alamkanak/weekview/WeekViewDrawingConfig.java]: You could encapsulate this logic in something like `config.getMinX()` and `config.getMaxX()`. That would make this method easier to parse.

File: library/src/main/java/com/alamkanak/weekview/WeekViewGestureHandler.java [hunk 4]
```diff
@@ -146,7 +146,22 @@ public boolean onScroll(MotionEvent e1, MotionEvent e2, float distanceX, float d
         switch (currentScrollDirection) {
             case LEFT:
             case RIGHT:
+                float minX = Integer.MIN_VALUE;
+                if (config.maxDate != null) {
```
Comment (thellmund) [inline library/src/main/java/com/alamkanak/weekview/WeekViewGestureHandler.java]: Same as in `WeekViewDrawingConfig`: Move this code to methods in `WeekViewConfig` and use `Math.min()` and `Math.max()`.

File: library/src/main/java/com/alamkanak/weekview/WeekViewGestureHandler.java [hunk 5]
```diff
@@ -193,8 +208,21 @@ private void onFlingHorizontal(float originalVelocityX) {
         final int velocityX = (int) (originalVelocityX * config.xScrollingSpeed);
         final int velocityY = 0;
 
-        final int minX = Integer.MIN_VALUE;
-        final int maxX = Integer.MAX_VALUE;
+        final int minX;
```
Comment (thellmund) [inline library/src/main/java/com/alamkanak/weekview/WeekViewGestureHandler.java]: Move to something like `config.getMinX()` and `config.getMaxX()`.

File: library/src/main/java/com/alamkanak/weekview/DateUtils.kt [hunk 6]
```diff
@@ -104,4 +105,8 @@ internal object DateUtils {
         }
     }
 
+    @JvmStatic
```
[the remark this answers is not in the document]
Comment (cs8898) [reply library/src/main/java/com/alamkanak/weekview/DateUtils.kt]: Moved to WeekViewConfig.java

File: library/src/main/java/com/alamkanak/weekview/WeekView.java [hunk 7]
```diff
@@ -1026,6 +1059,7 @@ public void goToDate(@NonNull Calendar date) {
 
         int diff = DateUtils.getDaysUntilDate(date);
 
+        //TODO ignore if already out of range
```
Comment (thellmund) [inline library/src/main/java/com/alamkanak/weekview/WeekView.java]: There’s still an open to-do.

File: library/src/main/java/com/alamkanak/weekview/WeekView.java [hunk 8]
```diff
@@ -1015,6 +1039,15 @@ public void goToCurrentTime() {
      * @param date The date to show.
      */
     public void goToDate(@NonNull Calendar date) {
+        //handle out of date range
```
Comment (thellmund) [inline library/src/main/java/com/alamkanak/weekview/WeekView.java]: Please state the rationale behind these two checks a little clearer.

File: library/src/main/java/com/alamkanak/weekview/WeekViewDrawingConfig.java [hunk 9]
```diff
@@ -29,6 +29,12 @@
     Paint todayHeaderTextPaint;
     private int currentAllDayEventHeight;
 
+
+    /**
+     * dates in the past have origin.x > 0
```
Comment (thellmund) [inline library/src/main/java/com/alamkanak/weekview/WeekViewDrawingConfig.java]: 👍

File: library/src/main/java/com/alamkanak/weekview/WeekView.java (hunk 7 above)
Comment (cs8898) [reply library/src/main/java/com/alamkanak/weekview/WeekView.java]: Yep ;D
currently i hadn't any good idea i will work on that later

Comment (cs8898) [reply library/src/main/java/com/alamkanak/weekview/WeekView.java]: Done, should be correct

**Comment by thellmund** [comment]
LGTM!

**Pull request merged**